.github/workflows/main.yml

```diff
@@ -55,13 +55,17 @@ jobs:
 
   test:
     runs-on: ubuntu-latest
+    strategy: 
+      matrix:
+        python: [py32, py39, py310, py311] 
+        pytorch: [torch20, torch21, torch22, torch23, torch24]
```

Comment:
Can we build this list dynamically? Suppose a new version of Pytorch appears, I would prefer to test/support it immediately, rather than having to manually update this list.

Something to try is: create a new job that populates this list of versions, then invokes the `test` job while passing in this list to "matrix".

See my other project: https://github.com/apoorvkh/ml-paper-template/blob/utils/verify/.github/workflows/push-main.yml

Reply:
Yep. Just trying to get it working, for now.

Reply:
`versions=($(curl -s https://pypi.org/pypi/torch/json | jq -r '.releases | keys[]' | grep -v '^1\.' | grep -E '\.[0]+$' | sort -rV | sed 's/\.0$//'))`